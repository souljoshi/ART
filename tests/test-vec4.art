int main()
{
    vec v1;
    vec v2;
    vec v3;
    vec v4;
    vec v5;
    vec v6;
    vec v7;
    vec v8;
    vec v9;
    vec v10;
    double f1;
    double f2;
    double f3;

    v2 = <66.5, 12.0>;
    v3 = <1.0, 55.1>;

    f1 = 10.5;
    f2 = 5.5;

    f3 = f1 + f2;

    //v4 = v2 + v3; // this is giving error right now, need to fix

    //v4 = <66.5, 12.0> + v3; // this is giving error right now, need to fix

    v4 = <66.5, 12.0> + <1.0, 55.1>;
    v5 = <66.5, 12.0> - <1.0, 55.1>;
    v6 = <66.5, 12.0> * <1.0, 55.1>;
    v7 = <66.5, 12.0> / <1.0, 55.1>;

    //v8 = <66.5, 12.0> % <1.0, 55.1>; 
    //will give error "Operation not supported for vectors" as expected

    v9 = <66.5, 12.0> * 2.0; // scalar mulitplication
    v10 = 2.0 * <66.5, 12.0>;

    v1[0] = 10.5;
    v1[1] = 4.2;

    //v3[0] = 9.9;

    printf(v1[0]);
    printf(v1[1]);

    //printf(v2[0]);
    //printf(v2[1]);

    //printf(v3[0]);
    //printf(v3[1]);

    printf(v4[0]);
    printf(v4[1]);
    printf(v5[0]);
    printf(v5[1]);
    printf(v6[0]);
    printf(v6[1]);
    printf(v7[0]);
    printf(v7[1]);
    printf(f3);
    printf(v9[0]);
    printf(v9[1]);
    printf(v10[0]);
    printf(v10[1]);
}