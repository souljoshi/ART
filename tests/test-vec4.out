10.500000
4.200000
67.500000
67.100000
65.500000
-43.100000
66.500000
661.200000
66.500000
0.217786
134.000000
728.300000
133.000000
24.000000
8911.000000
8739.600000
133.000000
24.000000
